From 36e6ed0cf2d106b12f737ba5429040df56753f05 Mon Sep 17 00:00:00 2001
From: Jeff Cooper <repoocaj@gmail.com>
Date: Fri, 20 Feb 2026 16:47:15 -0600
Subject: [PATCH 1/4] Allow multiple LittleFS partitions to be defined

---
 subsys/fs/littlefs_fs.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/subsys/fs/littlefs_fs.c b/subsys/fs/littlefs_fs.c
index 3d0c71e2367..17d1bc540a2 100644
--- a/subsys/fs/littlefs_fs.c
+++ b/subsys/fs/littlefs_fs.c
@@ -1072,8 +1072,8 @@ struct fs_mount_t FS_FSTAB_ENTRY(DT_DRV_INST(inst)) = { \
 	.fs_data = &fs_data_##inst, \
 	.storage_dev = (void *) \
 		COND_CODE_1(USE_PARTITION_MANAGER, \
-			    (COND_CODE_1(FIXED_PARTITION_EXISTS(littlefs_storage), \
-					 (FIXED_PARTITION_ID(littlefs_storage)), \
+			    (COND_CODE_1(FIXED_PARTITION_EXISTS(lfs_part_##inst), \
+					 (FIXED_PARTITION_ID(lfs_part_##inst)), \
 					 (FIXED_PARTITION_ID(storage)))), \
 			    (DT_FIXED_PARTITION_ID(FS_PARTITION(inst)))), \
 	.flags = FSTAB_ENTRY_DT_MOUNT_FLAGS(DT_DRV_INST(inst)), \
-- 
2.43.0

