From d7bb620e3c425a994282ae56766386a2f9f82c93 Mon Sep 17 00:00:00 2001
From: Jeff Cooper <repoocaj@gmail.com>
Date: Wed, 22 Apr 2026 11:14:52 -0500
Subject: [PATCH 2/2] Add ability to force file cycling on log_backend_fs

---
 subsys/logging/backends/log_backend_fs.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/subsys/logging/backends/log_backend_fs.c b/subsys/logging/backends/log_backend_fs.c
index 1f977403d65..dbda80ad1c6 100644
--- a/subsys/logging/backends/log_backend_fs.c
+++ b/subsys/logging/backends/log_backend_fs.c
@@ -4,6 +4,8 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+// clang-format off
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <zephyr/logging/log_backend.h>
@@ -34,6 +36,7 @@ static int get_log_file_id(struct fs_dirent *ent);
 #ifndef CONFIG_LOG_BACKEND_FS_TESTSUITE
 static uint32_t log_format_current = CONFIG_LOG_BACKEND_FS_OUTPUT_DEFAULT;
 #endif
+static volatile bool force_cycle = false;
 
 static int check_log_volume_available(void)
 {
@@ -164,6 +167,15 @@ int write_log_to_file(uint8_t *data, size_t length, void *ctx)
 
 	if (backend_state == BACKEND_FS_OK) {
 
+        /* Check if we need to create a new file. */
+        if (force_cycle) {
+			force_cycle = false;
+			rc = allocate_new_file(f);
+			if (rc < 0) {
+				goto on_error;
+			}
+		}
+
 		/* Check if new data overwrites max file size.
 		 * If so, create new log file.
 		 */
@@ -505,3 +517,9 @@ static const struct log_backend_api log_backend_fs_api = {
 LOG_BACKEND_DEFINE(log_backend_fs, log_backend_fs_api,
 		   IS_ENABLED(CONFIG_LOG_BACKEND_FS_AUTOSTART));
 #endif
+
+void log_backend_fs_force_cycle(void)
+{
+    /* Set the flag to force a file cycle. */
+    force_cycle = true;
+}
-- 
2.43.0

